sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala

```diff
@@ -107,4 +108,12 @@ trait AnalysisTest extends PlanTest {
          """.stripMargin)
     }
   }
+
+  protected def interceptParseException(
```

Comment:
Looks good but why don't we move this into `PlanTest` or `PlanTestBase` instead of changing inheritance hierarchy?

Reply:
It's because `ParseException` extends `AnalysisException`. And, all `AnalysisException`-related stuff resides in `AnalysisTest` which extends `PlanTest`. This is the right place to have this.

Reply:
Thanks.